File: Python/sentimental-mario-more/mario.py
def main():
    height = getsize()
    printgrid(height)


def getsize():
    while True:
        try:
            height = int(input("Height: "))
            if 1 <= height <= 8:
                break
            else:
                print("Enter a number 1-8 to proceed")
        except ValueError:
            print("Enter a number 1-8 to proceed")
    return height


def printgrid(height):
    block = "#"
    for i in range(height):
        space = " " * (height - 1 - i)
        print(f"{space}{block}  {block}")
        block += "#"


if __name__ == "__main__":
    main()

File: C/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int get_size(void);
void print_grid(int size);

// runs program
int main(void)
{
    // get block size
    int height = get_size();
    // print block size
    print_grid(height);
}
// ask user for block size
int get_size(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
    return height;
}
// prints mario blocks if input is 1-8
void print_grid(int height)
{
    char block[40] = "#";

    for (int x = 0; height > x; x++)
    {
        char space[40] = "";
        for (int j = (height - 1); x < j; j--)
        {
            strcat(space, " ");
        }
        printf("%s%s  %s", space, block, block);
        strcat(block, "#");
        printf("\n");
    }
}